WordPress/src/main/java/org/wordpress/android/ui/accounts/signup/SignupEpilogueFragment.kt

```diff
@@ -0,0 +1,901 @@
+                                AnalyticsTracker.track(Stat.ME_GRAVATAR_UPLOAD_EXCEPTION, properties)
+                                AppLog.e(AppLog.T.NUX, "Uploading image to Gravatar failed")
+                            }
+                        }
+                    }
+                }
+            } else {
+                ToastUtils.showToast(
+                    activity,
+                    R.string.error_locating_image,
+                    ToastUtils.Duration.SHORT
+                )
+            }
+        } else {
+            ToastUtils.showToast(activity, R.string.error_locating_image, ToastUtils.Duration.SHORT)
+        }
+    }
+
+    private fun undoChanges() {
+        mDisplayName = if (!TextUtils.isEmpty(mAccountStore.account.displayName)
+        ) mAccountStore.account.displayName else requireArguments().getString(ARG_DISPLAY_NAME)
+        mEditTextDisplayName.setText(mDisplayName)
+        mUsername = if (!TextUtils.isEmpty(mAccountStore.account.userName)
+        ) mAccountStore.account.userName else requireArguments().getString(ARG_USERNAME)
+        mEditTextUsername.setText(mUsername)
+        mInputPassword.editText.setText("")
+        updateAccountOrContinue()
+    }
+
+    private fun updateAccountOrContinue() {
+        if (changedUsername()) {
+            startProgressIfNeeded()
+            updateUsername()
+        } else if (changedDisplayName()) {
+            startProgressIfNeeded()
+            mIsUpdatingDisplayName = true
+            updateDisplayName()
+        } else if (changedPassword() && !mHasUpdatedPassword) {
+            startProgressIfNeeded()
+            mIsUpdatingPassword = true
+            updatePassword()
+        } else if (mSignupEpilogueListener != null) {
+            if (!mHasMadeUpdates) {
+                AnalyticsTracker.track(
+                    if (mIsEmailSignup
+                    ) Stat.SIGNUP_EMAIL_EPILOGUE_UNCHANGED
+                    else Stat.SIGNUP_SOCIAL_EPILOGUE_UNCHANGED
+                )
+            }
+            endProgressIfNeeded()
+            mSignupEpilogueListener!!.onContinue()
+        }
+    }
+
+    private fun updateDisplayName() {
+        val payload = PushAccountSettingsPayload()
+        payload.params = HashMap()
+        payload.params["display_name"] = mDisplayName
+        dispatcher.dispatch(AccountActionBuilder.newPushSettingsAction(payload))
+    }
+
+    private fun updatePassword() {
+        val payload = PushAccountSettingsPayload()
+        payload.params = HashMap()
+        payload.params["password"] = mInputPassword.editText.text.toString()
+        dispatcher.dispatch(AccountActionBuilder.newPushSettingsAction(payload))
+    }
+
+    private fun updateUsername() {
+        val payload = PushUsernamePayload(
+            (mUsername)!!, AccountUsernameActionType.KEEP_OLD_SITE_AND_ADDRESS
+        )
+        dispatcher.dispatch(AccountActionBuilder.newPushUsernameAction(payload))
```

Comment:
Though the current code matches the Java behaviour, I would suggest checking `mUsername` for `null` to avoid force unwrapping. Wdyt?
```suggestion
        mUsername?.let {
            val payload = PushUsernamePayload(it, AccountUsernameActionType.KEEP_OLD_SITE_AND_ADDRESS)
            dispatcher.dispatch(AccountActionBuilder.newPushUsernameAction(payload))
        }
```